fs_s3fs/_s3fs.py

```diff
@@ -705,10 +705,6 @@ def scandir(self, path, namespaces=None, page=None):
         _s3_key = self._path_to_dir_key(_path)
         prefix_len = len(_s3_key)
 
-        info = self.getinfo(path)
```

Comment:
mb we could add flag `strict` (true by default), and skip directory check if it is false?

Reply:
@vindex10 - 

From the project perspective:
S3 isn't a file system its only a key store, means when you add keys to it, most likely you don't explicitly create each "directory" key in the path.

From our perspective:
but than it breaks the `pyfilesystem2` abstraction.
and you will need to check is this `s3` add strict.

Reply:
So, let's stick to pyfilesystem2 abstraction. This is our choice to have weakened abstraction on top of s3.

Reply:
you are correct. better to use
```python
if self.strict:
```